File: Sp3/Problem1/H/src/Start.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Start {
    public static String[] tempMass;
    public static int posNow = 0;

    // Метод для нахождения общей площади участков
    public static void areaFind(SortedMap<Integer, Integer> pairMap) {

        for (Map.Entry<Integer, Integer> mapElem : pairMap.entrySet()) {

        }

    }

    public static void main(String[] args) throws IOException {

        // Открываем поток на чтение данных
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        // Кол-во пар обработанных участков
        int countPair = Integer.parseInt(reader.readLine());

        // map где храним участки
        SortedMap<Integer, Integer> pairMap = new TreeMap<Integer, Integer>();


        // Заполняем map
        for (int i = 0; i < countPair; i++) {
            tempMass = reader.readLine().split(" ");
            pairMap.put(Integer.parseInt(tempMass[0]), Integer.parseInt(tempMass[1]));
        }

    }
}

File: Sp3/Problem1/H/Start.py
# Кол-во пар
countPair = int(input())

# Массив пар
pairMap = list()

# Считывем пары
for elem in range(countPair):
    pairMap.append([int(i) for i in input().split()])

# отсортировали массив
pairMap.sort(key=lambda x: x[0])

# Позиция где мы находимся
posNow = 0
stepNow = 0

for elem in pairMap:

    # # Мы никуда не сдвигаемся, если концы остались такие же
    # # Так же является особым случаем для первого массива
    # if pairMap[posNow][1] == elem[1]:
    #     stepNow += 1
    #     continue
    if stepNow == 0:
        stepNow += 1
        continue

    # Если элемент расширяет границу
    # То добавим его конец к самому начальному и получим
    # Наибольшую границу
    if pairMap[posNow][1] >= elem[0]:
        if pairMap[posNow][1] < elem[1]:
            pairMap[posNow][1] = elem[1]

    else:  # Если цепочка прервалась напечатаем, что было получено ранее
        print(str(pairMap[posNow][0]) + " " + str(pairMap[posNow][1]))
        posNow = stepNow


    # Случай выхода из цикла
    # Не забудем напечатать последнюю пару
    if stepNow + 1 == len(pairMap):
        print(str(pairMap[posNow][0]) + " " + str(pairMap[posNow][1]))

    stepNow += 1

